src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -2980,6 +2981,9 @@ namespace QgsWms
       expressionContext << QgsExpressionContextUtils::layerScope( layer );
     expressionContext.setFeature( *feat );
 
+    QgsEditFormConfig editConfig { layer ? layer->editFormConfig() : QgsEditFormConfig() };
+    const bool honorFormConfig { layer && QgsServerProjectUtils::wmsFeatureInfoUseAttributeFormSettings( *mProject ) && editConfig.layout() == Qgis::AttributeFormLayout::DragAndDrop };
```

Comment:
genuinely naive question. It feels odd to test something that sounds like a purely view/UI detail "DragAndDrop" in a non-UI context. Perhaps that's just an odd naming. But looking at the enumeration it contains and the code below, I'm wondering if the code below couldn't also make sense for the AutoGenerated or UiFile case, since it is not obvious to me that QgsAttributeEditorContainer is correlated with DragAndDrop. Hope that makes sense... Anyway, feel free to ignore if that should make sense for readers familiar with those bits of code.

Reply:
@rouault QGIS server uses QGIS desktop to generate the project files (qgs/qgz) and QGIS desktop allow to store the fields configuration in a few different ways (drag and drop is one of those), not all these configuration options are available/implemented in the server (for instance the .ui file is not, probably because it is really very tied to a GUI context)   I don't know what we can do about it but yes, I agree it can look odd.